### PR TITLE
Fix Event recurrences duration and text in appointmentTemplate (T818393)

### PR DESCRIPTION
**Pull request by Olexi**

### CODE REVIEW

File: js/ui/scheduler/ui.scheduler.subscribes.js [hunk 1]
```diff
@@ -89,7 +89,6 @@ const subscribes = {
 
         let itemResources = this._resourcesManager.getResourcesFromItem(appointmentData);
         allDay = this.appointmentTakesAllDay(appointmentData) && this._workSpace.supportAllDayRow();
```
Comment (Krijovnick) [inline js/ui/scheduler/ui.scheduler.subscribes.js:91]: There is unnecessary changes

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js [hunk 2]
```diff
@@ -473,7 +473,10 @@ class BaseRenderingStrategy {
         if(viewStartDate.getTime() > endDate.getTime() || isRecurring) {
             var recurrencePartStartDate = position ? position.initialStartDate || position.startDate : realStartDate,
                 recurrencePartCroppedByViewStartDate = position ? position.startDate : realStartDate,
-                fullDuration = endDate.getTime() - realStartDate.getTime();
+                fullDuration =
+                viewStartDate.getTime() > endDate.getTime() ?
```
Comment (Krijovnick) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js:477]: There is no tests on this code

File: testing/tests/DevExpress.ui.widgets.scheduler/integration.recurringAppointments.tests.js [hunk 3]
```diff
@@ -1472,3 +1472,36 @@ QUnit.test("Appointment has correct occurrences dates with interval > 1, custom
...
+        dataSource: [{
+            text: "Recurrence",
+            startDate: new Date(2019, 2, 30, 2, 0),
+            startDateTimeZone: 'Europe/Copenhagen',
+            endDate: new Date(2019, 2, 30, 10, 0),
+            endDateTimeZone: 'Europe/Copenhagen',
+            recurrenceException: "",
+            recurrenceRule: "FREQ=DAILY"
+        }],
+        timeZone: 'Europe/Copenhagen',
```
Comment (Krijovnick) [inline testing/tests/DevExpress.ui.widgets.scheduler/integration.recurringAppointments.tests.js:1513]: Strange thing: the `timeZone` option is not work and that test work always before I change time zone in my computer. Is it a bag?

File: testing/tests/DevExpress.ui.widgets.scheduler/integration.recurringAppointments.tests.js [hunk 4]
```diff
@@ -1472,3 +1472,36 @@ QUnit.test("Appointment has correct occurrences dates with interval > 1, custom
...
+    var $firstAppointment = $(this.scheduler.appointments.getAppointment(0)),
+        firstAppointmentCoords = translator.locate($firstAppointment),
+        $thirdAppointment = $(this.scheduler.appointments.getAppointment(2)),
+        thirdAppointmentCoords = translator.locate($thirdAppointment);
+
+    assert.equal(firstAppointmentCoords.top, thirdAppointmentCoords.top, "Appointment first and third occurrences have same top coordinate");
+
+    assert.equal($firstAppointment.find(".dx-scheduler-appointment-content-date").eq(0).text(), $thirdAppointment.find(".dx-scheduler-appointment-content-date").eq(0).text(), "Appointment first and third occurrences have same start date text");
+    assert.equal($firstAppointment.find(".dx-scheduler-appointment-content-date").eq(2).text(), $thirdAppointment.find(".dx-scheduler-appointment-content-date").eq(2).text(), "Appointment first and third occurrences have same end date text");
+});
```
Comment (Krijovnick) [inline testing/tests/DevExpress.ui.widgets.scheduler/integration.recurringAppointments.tests.js:1542]: The last two tests are not necessary. They check the text, which was originally correct and there is no code for this.

**Review by andrewmakarov — DISMISSED** [review]

**Review by Krijovnick — DISMISSED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Review by Krijovnick — APPROVED** [review]
